### PR TITLE
Implement `java.lang.ProcessBuilder` for Windows

### PR DESCRIPTION
**Pull request by WojciechMazur**
This PR adds support for running sub-processes using ProcessBuilder on Windows.

Changes include: 
* Implementation of `WindowsProcess`
* Added bindings for used Windows API methods and structs
* Adjusted tests to allow for running tests on both Windows and JVM
* Fixed minor issues allowing to test whole `java.lang` package.

### CODE REVIEW

File: javalib/src/main/scala/java/lang/process/UnixProcess.scala [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import scala.scalanative.unsafe._
-import scala.scalanative.libc.{errno => err}
+import scala.scalanative.libc.{errno => err, signal => sig, _}
+import sig._
 import err.errno
 import scala.scalanative.posix.{
   fcntl,
-  pthread,
   signal,
+  pthread,
```
Comment (sjrd) [inline javalib/src/main/scala/java/lang/process/UnixProcess.scala:14]: Spurious change?

File: javalib/src/main/scala/java/lang/process/WindowsProcess.scala [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+      builder
+        .environment()
+        .entrySet()
+        .scalaOps
+        .toSeq
+        .map(e => s"${e.getKey()}=${e.getValue()}")
+    }.asInstanceOf[Ptr[Byte]]
+
+    val startupInfo = stackalloc[StartupInfoW]
+    zeroMemory(startupInfo)
```
Comment (sjrd) [inline javalib/src/main/scala/java/lang/process/WindowsProcess.scala:172]: Doesn't `stackalloc` already guarantee to return zeroed memory?

File: unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala [hunk 3]
```diff
@@ -7,30 +7,42 @@ import java.io.File
...
-      out.split("\n").toSet == Set("echo.sh", "err.sh", "hello.sh", "ls")
-    )
+    assertEquals(Scripts.values.map(_.filename), out.split(EOL).toSet)
   }
   @Test def execEnvp(): Unit = {
+    assumeFalse(
+      "Not possible in Windows, would use dir keyword anyway",
+      isWindows
+    )
+    assumeFalse("Not complient with JVM", executingInJVM)
```
Comment (sjrd) [inline unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala:33]:
```suggestion
    assumeFalse("Not compliant with JVM", executingInJVM)
```
Also in some other places.

**Review by sjrd — APPROVED** [review]
Only very minor comments.

File: javalib/src/main/scala/java/lang/process/WindowsProcess.scala (hunk 2 above)
Comment (WojciechMazur) [reply javalib/src/main/scala/java/lang/process/WindowsProcess.scala:172]: Yes, that is correct, I've forgotten about that. Explicit zeroing of memory is redundant.

**Pull request merged**